src/frontend/components/PlivoDialer.js

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import {FloatingActionButton, FontIcon} from 'material-ui';
+
+export default class PlivoDialer extends React.Component {
+  static propTypes = {
+    endpointUsername: React.PropTypes.string,
+    endpointPassword: React.PropTypes.string,
+    number: React.PropTypes.string
+  }
+
+  state = {
+    displayed: true,
+    plivoCallInProgress: false,
+    plivoStatusText: null
+  }
+
+  registerCallbacks() {
+    Plivo.onWebrtcNotSupported = () => {
+      this.setState({visible: false})
+    }
+    Plivo.onReady = () => {
+      console.log('PlivoJS is ready to be used')
+    }
+    Plivo.onMediaPermission = () => {
+      console.log('PlivoJS user media access permissions granted')
+    }
+    Plivo.onLogin = () => {
```

Comment:
I've implemented this fairly similarly to your mockups. A couple of divergences:

- You wanted the action button to be disabled while a call is in progress. That would prevent agents from hanging up calls at their discretion, which I imagine they might want to do in certain cases (ex: an abusive person on the other end; they've reached a fax machine that's now screeching at them).

- Transitioning the formatted phone number to light blue did indeed result in it getting lost visually. Instead, I'm applying a CSS transition to a blue shadow around the text as well as darkening the blue. Doesn't quite match the Material UI styling but it reads to me as activity.